internal/exec/util/file.go

```diff
@@ -177,9 +177,7 @@ func (u Util) PerformFetch(f *FetchOp) error {
 
 	defer func() {
 		tmp.Close()
-		if err != nil {
-			os.Remove(tmp.Name())
-		}
+		os.Remove(tmp.Name())
```

Comment:
this doesn't need to be `func(){}()` wrapped now, but otherwise LGTM. Maybe drop a comment explaining that we sometimes call it knowing it will fail and that's ok?

Reply:
Done, merging on green unless you say something.